airflow/jobs/scheduler_job.py

```diff
@@ -1440,94 +1436,89 @@ def _change_state_for_tasks_failed_to_execute(self, session=None):
 
         :param session: session for ORM operations
         """
-        if self.executor.queued_tasks:
-            TI = models.TaskInstance
-            filter_for_ti_state_change = (
-                [and_(
-                    TI.dag_id == dag_id,
-                    TI.task_id == task_id,
-                    TI.execution_date == execution_date,
-                    # The TI.try_number will return raw try_number+1 since the
-                    # ti is not running. And we need to -1 to match the DB record.
-                    TI._try_number == try_number - 1,  # pylint: disable=protected-access
-                    TI.state == State.QUEUED)
-                    for dag_id, task_id, execution_date, try_number
-                    in self.executor.queued_tasks.keys()])
-            ti_query = (session.query(TI)
-                        .filter(or_(*filter_for_ti_state_change)))
-            tis_to_set_to_scheduled = (ti_query
-                                       .with_for_update()
-                                       .all())
-            if len(tis_to_set_to_scheduled) == 0:
-                session.commit()
-                return
-
-            # set TIs to queued state
-            filter_for_tis = TI.filter_for_tis(tis_to_set_to_scheduled)
-            session.query(TI).filter(filter_for_tis).update(
-                {TI.state: State.SCHEDULED, TI.queued_dttm: None}, synchronize_session=False
-            )
+        if not self.executor.queued_tasks:
+            return
 
-            for task_instance in tis_to_set_to_scheduled:
-                self.executor.queued_tasks.pop(task_instance.key)
+        filter_for_ti_state_change = (
+            [and_(
+                TI.dag_id == dag_id,
+                TI.task_id == task_id,
+                TI.execution_date == execution_date,
+                # The TI.try_number will return raw try_number+1 since the
+                # ti is not running. And we need to -1 to match the DB record.
+                TI._try_number == try_number - 1,  # pylint: disable=protected-access
+                TI.state == State.QUEUED)
+                for dag_id, task_id, execution_date, try_number
+                in self.executor.queued_tasks.keys()])
+        ti_query = session.query(TI).filter(or_(*filter_for_ti_state_change))
+        tis_to_set_to_scheduled = ti_query.with_for_update().all()
+        if not tis_to_set_to_scheduled:
+            return
 
-            task_instance_str = "\n\t".join(
-                [repr(x) for x in tis_to_set_to_scheduled])
+        # set TIs to queued state
+        filter_for_tis = TI.filter_for_tis(tis_to_set_to_scheduled)
+        session.query(TI).filter(filter_for_tis).update(
+            {TI.state: State.SCHEDULED, TI.queued_dttm: None}, synchronize_session=False
+        )
 
-            session.commit()
```

Comment:
As I understand this could be removed, as commit will be executed anyway due to `@provide_session` decorator? But doesn't it happen only if `session` is not provided explicitly?

Reply:
You are correct, this method is restricted and called only in scheduler job.